plugin/src/main/groovy/com/novoda/staticanalysis/StaticAnalysisPlugin.groovy

```diff
@@ -39,12 +39,21 @@ class StaticAnalysisPlugin implements Plugin<Project> {
                                                           Task evaluateViolations) {
         NamedDomainObjectContainer<Violations> violationsContainer = pluginExtension.allViolations
         [
-                CheckstyleConfigurator.create(project, violationsContainer, evaluateViolations),
-                PmdConfigurator.create(project, violationsContainer, evaluateViolations),
-                FindbugsConfigurator.create(project, violationsContainer, evaluateViolations),
-                DetektConfigurator.create(project, violationsContainer, evaluateViolations),
-                KtlintConfigurator.create(project, violationsContainer, evaluateViolations),
-                LintConfigurator.create(project, violationsContainer, evaluateViolations)
+                CheckstyleConfigurator.create(project, violationsContainer, createTask('evaluateCheckstyleViolations', project, pluginExtension, evaluateViolations)),
+                PmdConfigurator.create(project, violationsContainer, createTask('evaluatePMDViolations', project, pluginExtension, evaluateViolations)),
+                FindbugsConfigurator.create(project, violationsContainer, createTask('evaluateFindbugsViolations', project, pluginExtension, evaluateViolations)),
+                DetektConfigurator.create(project, violationsContainer, createTask('evaluateDetektViolations', project, pluginExtension, evaluateViolations)),
+                KtlintConfigurator.create(project, violationsContainer, createTask('evaluateKtLintViolations', project, pluginExtension, evaluateViolations)),
+                LintConfigurator.create(project, violationsContainer, createTask('evaluateLintViolations', project, pluginExtension, evaluateViolations))
         ]
     }
+
+    private static Task createTask(String name, Project project, StaticAnalysisExtension extension, Task evaluateViolations) {
+        def task = project.tasks.create(name, EvaluateViolationsTask) { task ->
```

Comment:
Lately we integrated Gradle's lazy task configuration. To keep that, `register` needs to be configured. I introduced `TasksCompat` in order to make that backward compatible.

Reply:
These are not so expensive tasks. Maybe that can be fine. Instead of 1, we now create 6 tasks per module. But they're all small.